application/src/ext/java/org/opentripplanner/ext/fares/FaresToItineraryMapper.java

```diff
@@ -22,8 +25,12 @@ public static void addFaresToLegs(ItineraryFares fares, Itinerary i) {
 
     i.transformTransitLegs(leg -> {
       var legUses = fares.getLegProducts().get(leg);
-      leg.setFareProducts(ListUtils.combine(itineraryFareUses, legUses));
-      return leg;
+      var allUses = ListUtils.combine(itineraryFareUses, legUses);
```

Comment:
Same comment as for the Alerts, maybe use `interface ProductAware`

Reply:
I've introduced an interface but I've attached it to TransitLeg (for now). Let's discuss if that is good enough.